docs/src/main/asciidoc/software-transactional-memory.adoc

```diff
@@ -8,22 +8,79 @@ https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+to a relational database which supports https://pubs.opengroup.org/onlinepubs/009680699/toc.pdf[the X/Open XA
+standard].
+
+Note, the STM implementation provided by Quarkus is based on the https://narayana.io/docs/project/index.html#d0e16066[Narayana STM] implementation. This document isn't meant to be a replacement for that project's documentation so you may want
+to look at that for more detail. However, we will try to focus more on how you can combine some of the key capabilities
+into Quarkus when developing Kubernetes native applications and microservices.
+
+== Why use STM with Quarkus?
+
+Now you may still be asking yourself "Why STM instead of JTA?" or "What are the benefits
+to STM that I don't get from JTA?" Let's try to answer those or similar questions, with
+a particular focus on why we think they're great for Quarkus, microservices and Kubernetes
+native applications. So in no specific order ...
+
+* The goal of STM is to simplify object reads and writes from multiple threads/protect
+state from concurrent updates. The Quarkus STM implementation will safely manage any conflicts between
+these threads using whatever isolation model has been chosen to protect that specific state
+instance (object in the case of Quarkus). In Quarkus STM, there are two isolation implementations,
+pessimistic (the default), which would cause conflicting threads to be blocked until the original
+has completed its updates (committed or aborted the transaction); then there's the optimistic
+approach which allows all of the threads to proceed and checks for conflicts at commit time, where
+one or more of the threads may be forced to abort if there have been conflicting updates.
+
+* STM objects have state but it doesn't need to be persistent (durable). In fact the
+default behaviour is for objects managed within transactional memory to be volatile, such that
+if the service or microservice within which they are being used crashes or is spawned elsewhere, e.g.,
+by a scheduler, all state in memory is lost and the objects start from scratch. But surely you get this and more
+with JTA (and a suitable transactional datastore) and don't need to worry about restarting your application?
+Not quite. There's a trade-off here: we're doing away
+with persistent state and the overhead of reading from and then writing (and sync-ing) to the datastore during each
+transaction. This makes updates to (volatile) state very fast but you still get the benefits of atomic updates
+across multiple STM objects (e.g., objects your team wrote then calling objects you inherited from another team and requiring
+them to make all-or-nothing updates), as well as consistency
+and isolation in the presence of concurrent threads/users (common in distributed microservices architectures).
+Furthermore, not all stateful applications need to be durable - even when JTA transactions are used, the tend to be the
```

Comment:
```suggestion
Furthermore, not all stateful applications need to be durable - even when JTA transactions are used, it tends to be the
```
but it might just be me not being native.

Reply:
Good catch.